src/components/structures/auth/ForgotPassword.tsx

```diff
@@ -174,24 +178,28 @@ export default class ForgotPassword extends React.Component<IProps, IState> {
             return;
         }
 
-        Modal.createTrackedDialog('Forgot Password Warning', '', QuestionDialog, {
-            title: _t('Warning!'),
-            description:
-                <div>
-                    { _t(
-                        "Changing your password will reset any end-to-end encryption keys " +
-                        "on all of your sessions, making encrypted chat history unreadable. Set up " +
-                        "Key Backup or export your room keys from another session before resetting your " +
-                        "password.",
-                    ) }
-                </div>,
-            button: _t('Continue'),
-            onFinished: (confirmed) => {
-                if (confirmed) {
-                    this.submitPasswordReset(this.state.email, this.state.password);
-                }
-            },
-        });
+        if (this.state.logout_devices) {
+            Modal.createTrackedDialog('Forgot Password Warning', '', QuestionDialog, {
+                title: _t('Warning!'),
+                description:
+                    <div>
+                        { _t(
+                            "Signing out all sessions will reset the end-to-end encryption keys " +
+                            "making encrypted chat history unreadable. Set up " +
+                            "Key Backup or export your room keys from another session before resetting your " +
+                            "password.",
```

Comment:
Could we use this opportunity to change this message so it's less misleading? It talks about *resetting* end-to-end encryption keys, which makes it sound like something will happen server-side regarding this.

But in truth, it's the clients willingly *deleting* their own keys once they notice they've been signed out. So perhaps something like:

```suggestion
                            "Signing out your sessions will delete the message encryption keys stored on " +
                            "them, making encrypted chat history unreadable. Set up Key Backup or export " +
                            "your message keys from one of your sessions before proceeding.",
```

(Aside: also, "sessions" vs "devices"? Did we settle on one of those? I feel like we're still using them interchangeably all over the place.)

Reply:
The last time @amshakal and I talked about it for the device list view, we decided "devices" was the least confusing word to use, but I don't think there's broad consensus on this beyond the Crypto Team.

Reply:
FWIW, that's my take on this as well.